src/sampler_learning.py

```diff
@@ -5,22 +5,36 @@
 import numpy as np
 from predicators.src.structs import ParameterizedOption, LiftedAtom, Variable, \
     Object, Array, State, _Option, Datastore, STRIPSOperator, OptionSpec, \
-    NSRTSampler
+    NSRTSampler, NSRT
 from predicators.src import utils
 from predicators.src.torch_models import MLPClassifier, NeuralGaussianRegressor
 from predicators.src.settings import CFG
+from predicators.src.envs import create_env
+from predicators.src.approaches.oracle_approach import get_gt_nsrts
 
 
 def learn_samplers(strips_ops: List[STRIPSOperator],
                    datastores: List[Datastore], option_specs: List[OptionSpec],
                    sampler_learner: str) -> List[NSRTSampler]:
     """Learn all samplers for each operator's option parameters."""
     samplers = []
+    if sampler_learner == "oracle":
+        env = create_env(CFG.env)
+        # We don't need to match ground truth NSRTs with no
+        # continuous parameters, so filter them out.
+        gt_nsrts = {
+            nsrt
+            for nsrt in get_gt_nsrts(env.predicates, env.options)
+            if nsrt.option.params_space.shape != (0, )
```

Comment:
remove space?

Reply:
hah, you wish. we're at the mercy of YAPF now!